pkg/test/verify/cluster.go

```diff
@@ -25,12 +25,12 @@ func AddToolchainClusterAsMember(t *testing.T, functionToVerify FunctionToVerify
 	defer gock.Off()
 	status := test.NewClusterStatus(toolchainv1alpha1.ToolchainClusterReady, corev1.ConditionTrue)
 	memberLabels := []map[string]string{
-		Labels("", "", test.NameHost),
-		Labels(cluster.Member, "", test.NameHost),
-		Labels(cluster.Member, "member-ns", test.NameHost)}
+		Labels("", test.NameHost),
+		Labels("member-ns", test.NameHost)}
 	for _, labels := range memberLabels {
 
 		t.Run("add member ToolchainCluster", func(t *testing.T) {
+			labels["namespace"] = test.HostOperatorNs
```

Comment:
this line 
```go
labels["namespace"] = test.HostOperatorNs
```
as well as the other one
```go
labels["namespace"] = ""
```
in a combination with 
```go
	memberLabels := []map[string]string{
		Labels("", test.NameHost),
		Labels("member-ns", test.NameHost)}
	for _, labels := range memberLabels {
```
doesn't make much sense. At first you create two types of labels (either with or without the namespace set) that you iterate over and then you rewrite the configured namespace in the labels which overrides the preconfigured ones. 
Please do the first or the latter, don't combine them - either drop the for preconfigured array and the `for` loop use some if-else based on the value in the label: 
```go
		t.Run("add member ToolchainCluster", func(t *testing.T) {
			for _, withCA := range []bool{true, false} {
				toolchainCluster, sec := test.NewToolchainCluster("east", "secret", status, labels)
				if withCA {
					toolchainCluster.Spec.CABundle = "ZHVtbXk="
				}
				cl := test.NewFakeClient(t, toolchainCluster, sec)
				service := newToolchainClusterService(t, cl, withCA)
				defer service.DeleteToolchainCluster("east")

				// when
				err := functionToVerify(toolchainCluster, cl, service)

				// then
				if labels["namespace"] != "" {
					require.NoError(t, err)
					cachedToolchainCluster, ok := cluster.GetCachedToolchainCluster("east")
					require.True(t, ok)

					assert.Equal(t, labels["namespace"], cachedToolchainCluster.OperatorNamespace)

					// check that toolchain cluster role label tenant was set only on member cluster type
					require.NoError(t, cl.Get(context.TODO(), client.ObjectKeyFromObject(toolchainCluster), toolchainCluster))
					assert.Equal(t, status, *cachedToolchainCluster.ClusterStatus)
					assert.Equal(t, test.NameHost, cachedToolchainCluster.OwnerClusterName)
					assert.Equal(t, "http://cluster.com", cachedToolchainCluster.APIEndpoint)
				} else {
					require.Error(t, err)
					_, ok := cluster.GetCachedToolchainCluster("east")
					require.False(t, ok)
				}
			}
		})
```

Reply:
Okay right